allure-generator/src/main/java/io/qameta/allure/retry/RetryPlugin.java

```diff
@@ -47,26 +49,26 @@ public void aggregate(final Configuration configuration,
 
     private Consumer<TestResult> addRetries(final List<TestResult> results) {
         return latest -> {
-            final List<RetryItem> retries = new ArrayList<>();
-            latest.addExtraBlock(RETRY_BLOCK_NAME, retries);
-            results.stream()
+            final List<RetryItem> retries = results.stream()
                     .sorted(byTime())
                     .filter(result -> !latest.equals(result))
-                    .forEach(addRetry(latest, retries));
+                    .map(retry -> retry.withHidden(true))
+                    .map(retry -> new RetryItem()
+                            .withStatus(retry.getStatus())
+                            .withStatusDetails(retry.getStatusDetailsSafe().getMessage())
+                            .withTime(retry.getTime())
+                            .withUid(retry.getUid()))
+                    .collect(Collectors.toList());
+            latest.addExtraBlock(RETRY_BLOCK_NAME, retries);
+            final Set<Status> statuses = retries.stream()
+                    .map(RetryItem::getStatus)
+                    .distinct()
+                    .collect(Collectors.toSet());
 
-        };
-    }
+            statuses.remove(Status.PASSED);
+            statuses.remove(Status.SKIPPED);
 
-    private Consumer<TestResult> addRetry(final TestResult latest, final List<RetryItem> retries) {
-        return retried -> {
-            retried.setHidden(true);
-            retried.getStatusDetailsSafe().setFlaky(true);
-            retries.add(new RetryItem()
-                    .withUid(retried.getUid())
-                    .withStatus(retried.getStatus())
-                    .withTime(retried.getTime())
-            );
-            latest.getStatusDetailsSafe().setFlaky(true);
+            latest.getStatusDetailsSafe().setFlaky(!statuses.isEmpty());
```

Comment:
Currently if a test fails 100% times, it will be displayed flacky. 
It is also wrong, I think